### PR TITLE
Updates for 2024 Wfs (PR Testing, IBs, Patatrack, inputs)

### PR DESCRIPTION
**Pull request by AdrianoDee**
#### PR description:

This PR updates the wfs used in PR testing, IB test (standard and GPU).  It includes https://github.com/cms-sw/cmssw/commit/f77b3b3bd417e0ce73302d519b8ca1dfb8aa5078 from @mmusich. 

For the moment the 2023 wfs are kept as they are with the idea of reducing them later this year.

The default input GEN-SIM datasets for 2024 conditions are updated to `14_0_0_pre3` RelVals.

### CODE REVIEW

**Comment by AdrianoDee** [comment]
enable gpu

**Comment by AdrianoDee** [comment]
please test

**Comment by AdrianoDee** [comment]
please test

**Comment by AdrianoDee** [comment]
code-checks

**Comment by AdrianoDee** [comment]
please test

File: Configuration/PyReleaseValidation/scripts/runTheMatrix.py [hunk 1]
```diff
@@ -78,14 +78,15 @@ def runSelected(opt):
...
                     # Run3
-                    11634.0,    # TTbar_14TeV 
-                    13234.0,    # RelValTTbar_14TeV             FastsSim
-                    12434.0,    # RelValTTbar_14TeV
-                    12446.0,    # RelValZEE_13
-                    12634.0,    # RelValTTbar_14TeV             PU = Run3_Flat55To75_PoissonOOTPU
-                    12434.7,    # RelValTTbar_14TeV             mkFit
+                    11634.0,    # TTbar_14TeV                   2021
+                    13234.0,    # RelValTTbar_14TeV             2021 FastsSim
+                    12434.0,    # RelValTTbar_14TeV             2023
```
Comment (mmusich) [inline Configuration/PyReleaseValidation/scripts/runTheMatrix.py:83]: any strong reason to keep this in the system?

Comment (mmusich) [reply Configuration/PyReleaseValidation/scripts/runTheMatrix.py:83]: OK, just for the record, this is what is causing the changes explained at https://github.com/cms-sw/cmssw/pull/44026#issuecomment-1963539080.

**Comment by AdrianoDee** [comment]
+pdmv

File: Configuration/PyReleaseValidation/scripts/runTheMatrix.py (hunk 1 above)
Comment (srimanob) [reply Configuration/PyReleaseValidation/scripts/runTheMatrix.py:83]: Hi,
I think we should keep at least one workflow around, to monitor its healthy if we will have legacy wf later (not hlt part, of course).

Comment (mmusich) [reply Configuration/PyReleaseValidation/scripts/runTheMatrix.py:83]: > I think we should keep at least one workflow around, to monitor its healthy if we will have legacy wf later

not proposing to remove entirely btw, just in the limited matrix.

**Comment by mmusich** [comment]
urgent

**Comment by mmusich** [comment]
@cms-sw/upgrade-l2 can you please have a look?

**Comment by srimanob** [comment]
+Upgrade

**Comment by antoniovilela** [comment]
+1

**Pull request merged**